purefa/README.md

```diff
@@ -0,0 +1,130 @@
+# Agent Check: PureFA
+
+## Overview
+
+This check monitors the [Pure Storage FlashArray][3] through the [Datadog Agent][2] and the [Pure Storage Prometheus exporter][1]. 
+
+The integration can provide performance data at the array, host, volume and pod level, as well as high-level capacity and configuration information.
+
+Monitor multiple FlashArrays and aggregate these into a single dashboard, or group together by custonmer defined environment.
+
+Note: This integration requires:
+
+ - OpenMetricsBaseCheckV2 which is available in Agent v7.26.x+
+ - Python 3
+ - The [Pure Storage Prometheus exporter][1] is installed and running in a containerized environment. Refer to the [GitHub repo][1] for installation instructions.
+
+## Setup
+
+Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the Autodiscovery Integration Templates for guidance on applying these instructions.
+
+### Installation
+
+1. [Download and launch the Datadog Agent][9].
+2. Manually install the Pure FlashArray integration. See [Use Community Integrations][10] for more details based on your environment.
+
+
+#### Host
+
+To configure this check for an Agent running on a host, run `datadog-agent integration install -t datadog-purafa==<INTEGRATION_VERSION>`.
+
+### Configuration
+
+1. Create a local user on your FlashArray with the Read-Only role and generate an API token for this user.
+   ![Generate an API Key](https://raw.githubusercontent.com/DataDog/integrations-extras/master/purefa/images/API.png) 
+2. Add the following configuration block to the `purefa.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your purefa performance data. See the sample [purefa.d/conf.yaml][4] for all available configuration options.
+
+Note: The `/array` endpoint is required as an abolute minimum when creating your configuration file.
+
+```yaml
+init_config:
+   timeout: 60
+
+instances:
+
+  - openmetrics_endpoint: http://<exporter_ip_or_fqdn>:9491/metrics/flasharray/array?endpoint=<array_ip_or_fqdn>
+    tags:
+       - env:<env>
+       - fa_array_name:<full_fqdn>
+       - host:<full_fqdn>
+    hostname_label: host
+    hostname_format: <HOSTNAME>.<domain_name>
+    headers:
+       Authorization: Bearer <api_token>
+    min_collection_interval: 120
+    empty_default_hostname: true
+    max_returned_metrics: 100000
```

Comment:
```suggestion
```
These are optional and not necessary to include in every set up, so it would be better to remove them from the README.

Reply:
The way our integration works it is better to have these configured this way otherwise strange things can start to happen. for example the api_token is absolutely required for authentication and the max_returned_metrics must be this high to ensure we don't truncate data for very large and heavily configured arrays

Reply:
@yzhan289 are you happy to let us keep this in the README given out requirements?

Reply:
I think the others are fine, although the `max_returned_metrics: 100000` part is something I'll need to check with the team, since it is a hidden config option meant to be used for support cases.

Reply:
That one is actually crucial when we are dealing with enterprise class arrays. The default is way to small

Reply:
Thanks. In that case, can you have the config enabled and uncommented in the `conf.yaml`?

Reply:
@yzhan289 can you please re-review this PR to clear your change request?